### PR TITLE
Finished battles README #839

### PR DESCRIPTION
**Pull request by lczielinski**
Finished src/battles README with a brief overview of all battle src and header files. Our task was to go through the existing battle code and update the wiki page, and after coming across a blank README we decided to fill it in with a high-level overview of each relevant .h and .c file.

### CODE REVIEW

**Review by Elsa-Athiley — APPROVED** [review]
Everything looks good to me

**Review by adinilfeld — APPROVED** [review]
Looks great!

File: src/battle/README.md [hunk 1]
```diff
@@ -1 +1,21 @@
 Battle State README.md
```
[the remark this answers is not in the document]
Comment (lczielinski) [reply src/battle/README.md]: Got it, I made these style changes

**Review by firatciftci — APPROVED** [review]
Looks great!

**Comment by firatciftci** [comment]
Issue Score: ✔️+

Comments:
Good work overall! The only feedback I have is that, for future pull requests, you should include a final comment summarizing what was included in the PR after it gets merged into `dev`.

**Pull request merged**